WMF Framework/SharedContainerCache.swift

```diff
@@ -1,19 +1,14 @@
 import Foundation
 
-public final class SharedContainerCache<T: Codable> {
-    
-    // values must all be distinct
-    public enum PathComponent: String {
-        case widgetCache = "Widget Cache"
-        case pushNotificationsCache = "Push Notifications Cache"
-        case userDataExportSyncInfo = "User Data Export Sync Info"
-    }
-    
-    private let pathComponent: PathComponent
+public final class SharedContainerCache<T: Codable>: SharedContainerCacheHousekeepingProtocol {
```

Comment:
Optional suggestion: For common file names, you could set up a struct (separate from the class unfortunately, because it's generic):

```
struct SharedContainerCacheCommonFileNames {
    static let pushNotificationsCache = "Push Notifications Cache"
}

public final class SharedContainerCache<T: Codable>: ...
```

This way you don't have multiple string literals peppered throughout the codebase. A typo on any of them could risk throwing things off with caching.

```
let sharedCache = SharedContainerCache<PushNotificationsCache>(fileName: SharedContainerCacheCommonFileNames.pushNotificationsCache, defaultCache: { PushNotificationsCache(settings: .default, notifications: []) })
```

Reply:
Done!